### PR TITLE
Feature 13  integrate salaries chart with backend - Issue #13

### LINKED ISSUE #13: Feature - integrate salaries chart with backend
## Checklist ✅

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/Platzi-Master-C8/gethired-base/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

The chart should update accordingly to the data received from the backend

## Requirements 🌈
To be connected to the real backend
to update the chart in the calculator and comparison tabs
To work with redux toolkit

### PR DESCRIPTION
**Pull request by johannpereze**
## Checklist

- [X] I have followed the [contributor guideline](https://github.com/Platzi-Master-C8/gethired-base/blob/main/CONTRIBUTING.md)
- [ ] Tests are included
- [ ] Documentation is changed or added
- [X] Related issue has been created
- [X] Commits messages follows [commit guideline](https://github.com/Platzi-Master-C8/gethired-base/blob/main/CONTRIBUTING.md/#Commits)

### Type of PR

- [x] Feature
- [ ] Bugfix
- [ ] Hotfix
- [ ] Chore

### Merge strategy

- [x] Squash (default)
- [ ] Merge commit (branch merges, usually used for merges to master)
- [ ] Rebase (preserve individual commits without merge commit)

### Purpose

The chart should update accordingly to the data received from the backend
> Enlist the purposes or write an abastract about it, please be detailed, remember that not all the members of the team have the context of yout tasks
### Relevant tickets

> Paste here the link to the task ticket
resolves [#13 ]

### Related PR's

> Paste here the links to related PR's
### Screenshots

> Add screenshots that could help others do understand even more the objective of your pr

### CODE REVIEW

File: src/app/CalculateSalary/selectors.js [hunk 1]
```diff
@@ -4,3 +4,12 @@ export const selectFormMain = createSelector(
     (state) => state.Calculate.formMain,
     (form) => form,
 );
```
Comment (yadurani) [inline src/app/CalculateSalary/selectors.js:6]: Please add in title a message for describing PR

File: src/app/CalculateSalary/slice.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { mockDataProfile } from "Constants/mockData";
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js]: Change for single quotes

File: src/app/CalculateSalary/slice.js [hunk 3]
```diff
@@ -9,8 +11,20 @@ const initialState = {
...
 };
 
+
+export const fetchChartData = createAsyncThunk("post/fetchChartData", (profile) => {
+    return getSalaryProfile('salaries', profile)
+}
+)
+export const fetchComparisonChartData = createAsyncThunk("post/fetComparisonchChartData", async () => // TODO: receive the 2 profiles as arguments when we develop the state for the comparison forms
+    [await getSalaryProfile('salaries', mockDataProfile), await getSalaryProfile('salaries', mockDataProfile)]
+
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js:22]: Remove this space

File: src/app/CalculateSalary/slice.js [hunk 4]
```diff
@@ -9,8 +11,20 @@ const initialState = {
...
         location: '',
     },
+    chartData: [],
+    comparisonChartData: [],
 };
 
+
+export const fetchChartData = createAsyncThunk("post/fetchChartData", (profile) => {
+    return getSalaryProfile('salaries', profile)
+}
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js]: Please tabulate these lines

File: src/app/CalculateSalary/slice.js [hunk 5]
```diff
@@ -28,6 +42,14 @@ const calculateSalary = createSlice({
             state.formMain.technologies = state.formMain.technologies.filter((chip) => chip !== action.payload);
         },
     },
+    extraReducers: {
+        [fetchChartData.fulfilled]: (state, action) => {
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js:46]: Please add comment with TODO: Add stages when api is rejected

File: src/components/TabCalculate/TabCalculate.jsx [hunk 6]
```diff
@@ -8,14 +8,15 @@ import Button from '@mui/material/Button';
...
 
-import { values1, currencyName } from 'Constants';
+import { currencyName } from 'Constants';
 
-import { changesForm, clearFormMain, deleteChip } from 'App/CalculateSalary/slice';
+import { changesForm, clearFormMain, deleteChip, fetchChartData } from 'App/CalculateSalary/slice';
 import { selectFormMain } from 'App/CalculateSalary/selectors';
 
 import { disabled } from 'Helpers';
+import { selectChartData } from '../../app/CalculateSalary/selectors';
```
Comment (yadurani) [inline src/components/TabCalculate/TabCalculate.jsx]: Delete line, add selectedChartData in line 14, this duplicate route

File: src/components/TabCalculate/TabCalculate.jsx [hunk 7]
```diff
@@ -66,17 +67,44 @@ TabCalculate.propTypes = {
...
+        is_remote: PropTypes.bool,
+        location: PropTypes.string,
+        title_id: PropTypes.string,
+        technologies: PropTypes.arrayOf(PropTypes.string),
+    }).isRequired,
+    addChartData: PropTypes.func.isRequired,
+    chartData: PropTypes.arrayOf(
+        PropTypes.shape({
+            average: PropTypes.number,
+            top: PropTypes.number,
```
Comment (yadurani) [inline src/components/TabCalculate/TabCalculate.jsx:81]: the bottom proptypes is missing?

File: src/components/TabCompare/TabCompare.jsx [hunk 8]
```diff
@@ -8,9 +10,11 @@ import { Switch } from '@master-c8/icons';
 import FormCard from 'Components/FormCard';
 import NormalDistributionChart from 'Components/Charts';
 
-import { initialValues, values2, currencyName } from 'Constants';
+import { initialValues, currencyName } from 'Constants';
+import { fetchComparisonChartData } from '../../app/CalculateSalary/slice';
```
Comment (yadurani) [inline src/components/TabCompare/TabCompare.jsx]: Changes this routes for App/CalculateSalary/slice

File: src/components/TabCompare/TabCompare.jsx [hunk 9]
```diff
@@ -8,9 +10,11 @@ import { Switch } from '@master-c8/icons';
 import FormCard from 'Components/FormCard';
 import NormalDistributionChart from 'Components/Charts';
 
-import { initialValues, values2, currencyName } from 'Constants';
+import { initialValues, currencyName } from 'Constants';
+import { fetchComparisonChartData } from '../../app/CalculateSalary/slice';
+import { selectComparisonChartData } from '../../app/CalculateSalary/selectors';
```
Comment (yadurani) [inline src/components/TabCompare/TabCompare.jsx]: Changes this routes for App/CalculateSalary/selectors

File: src/components/TabCompare/TabCompare.jsx [hunk 10]
```diff
@@ -72,18 +80,47 @@ const TabCompare = () => {
...
     );
 };
 
-export default TabCompare;
+TabCompare.propTypes = {
+    addChartData: PropTypes.func.isRequired,
+    comparisonChartData: PropTypes.arrayOf(
+        PropTypes.shape({
+            average: PropTypes.number,
+            top: PropTypes.number,
```
Comment (yadurani) [inline src/components/TabCompare/TabCompare.jsx:102]: The bottom props is missing?

File: src/components/TabCompare/TabCompare.jsx [hunk 11]
```diff
@@ -72,18 +80,47 @@ const TabCompare = () => {
...
+        }),
+    ),
+};
+
+TabCompare.defaultProps = {
+    comparisonChartData: [
+        {
+            average: 0,
+            top: 0,
+            number: 0,
```
Comment (yadurani) [inline src/components/TabCompare/TabCompare.jsx]: I don't understand why the default values are different from proptypes?
`PropTypes.shape({
            average: PropTypes.number,
            top: PropTypes.number,
            is_remote: PropTypes.number,
}`

default props

`{
            average: 0,
            top: 0,
            number: 0,
}`

**Review by yadurani — CHANGES_REQUESTED** [review]

File: src/app/CalculateSalary/selectors.js (hunk 1 above)
Comment (johannpereze) [reply src/app/CalculateSalary/selectors.js:6]: done

File: src/app/CalculateSalary/slice.js [hunk 12]
```diff
@@ -9,8 +11,20 @@ const initialState = {
         is_remote: false,
         location: '',
     },
+    chartData: [],
+    comparisonChartData: [],
 };
 
+
+export const fetchChartData = createAsyncThunk("post/fetchChartData", (profile) => {
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js]: Changes "post/fetchChartData" for 'post/fetchChartData'

File: src/app/CalculateSalary/slice.js [hunk 13]
```diff
@@ -9,8 +11,20 @@ const initialState = {
...
+    return getSalaryProfile('salaries', profile)
+}
+)
+export const fetchComparisonChartData = createAsyncThunk("post/fetComparisonchChartData", async () => // TODO: receive the 2 profiles as arguments when we develop the state for the comparison forms
```
Comment (yadurani) [inline src/app/CalculateSalary/slice.js]: changes "post/fetComparisonchChartData" for 'post/fetComparisonchChartData'

**Review by yadurani — CHANGES_REQUESTED** [review]

**Review by yadurani — APPROVED** [review]
LGTM

**Pull request merged**